### PR TITLE
Add tests for performable moves and fix some errors in it

### PR DESCRIPTION
**Pull request by ArcherTheReal**
# Changelog
- add 2 unit tests
- add 1 random test
- fix that shift left and right gets added as performable moves if `self._braid < 2`

### CODE REVIEW

File: test/test_braid.py [hunk 1]
```diff
@@ -512,3 +513,81 @@ def test_remove_sigma_inverse_pair_indices4(self):
...
+        performable_moves.append(braid.destabilization)
+        
+        conjugation_values = list(range(-braid._n+1,0)) + list(range(1,braid._n))
+        conjugation_indices = (range(0, braid._braid.shape[0] + 2))
+        conjugation_performable_moves = []
+        for v in conjugation_values:
+            for i in conjugation_indices:
+                conjugation_performable_moves = conjugation_performable_moves + [partial(braid.conjugation,value=v,index=i)]
+
+        performable_moves = performable_moves + conjugation_performable_moves
```
Comment (Error-42) [inline test/test_braid.py:538]: use `+=` (applies later as well)

**Review by Error-42 — CHANGES_REQUESTED** [review]

File: test/test_braid.py [hunk 2]
```diff
@@ -512,3 +513,81 @@ def test_remove_sigma_inverse_pair_indices4(self):
...
+            move()
+    
+    def get_all_moves(self, braid):
+        performable_moves = [braid.shift_left,braid.shift_right,partial(braid.stabilization,inverse=True), partial(braid.stabilization,inverse=False)] #Always performable
+
+        performable_moves.append(braid.destabilization)
+        
+        conjugation_values = list(range(-braid._n+1,0)) + list(range(1,braid._n))
+        conjugation_indices = (range(0, braid._braid.shape[0] + 2))
+        conjugation_performable_moves = []
```
Comment (Error-42) [inline test/test_braid.py:533]: If possible fix mypy error

> `test\test_braid.py:533: error: Need type annotation for "conjugation_performable_moves" (hint: "conjugation_performable_moves: list[<type>] = ...")  [var-annotated]`

**Review by Error-42 — CHANGES_REQUESTED** [review]

File: test/test_braid.py [hunk 3]
```diff
@@ -569,14 +570,18 @@ def test_performable_moves(self, sigmas):
...
+
+        all_states = []       
         for move in all_moves:
             try:
                 state = move()
             except IllegalTransformationException as e:
                 pass
             else:
-                assert state in states
+                all_states.append(state)
```
Comment (Error-42) [inline test/test_braid.py:581]: `assert state in states` shouldn't have been removed.

File: test/test_braid.py [hunk 4]
```diff
@@ -569,14 +570,18 @@ def test_performable_moves(self, sigmas):
...
+        assert len(states) == len(all_states)
+        for state in states:
+            assert state in all_states
```
Comment (Error-42) [inline test/test_braid.py:584]: This fix should've been applied to `test_performable_moves_random` as well. Maybe remove code duplication so this type of mistake can't be made in the future?

**Review by Error-42 — CHANGES_REQUESTED** [review]

File: test/test_braid.py [hunk 5]
```diff
@@ -521,3 +523,85 @@ def test_remove_sigma_inverse_pair_indices4(self):
...
+            except IllegalTransformationException as e:
+                pass
+            else:
+                all_states.append(state)
+        assert len(states) == len(all_states)
+        for state in states:
+            assert state in all_states
+    @pytest.mark.parametrize("low, high, size", [[0, 1, 10], [-5,5,1], [-1, 1, 10], [-5, 5, 15]])
+    def test_performable_moves_random(self, low, high, size):
+        rng = np.random.default_rng(42)
```
Comment (fhuszar) [inline test/test_braid.py:587]: Never use random numbers in unit tests. If you do, make sure to create a randomStream with a fixed seed for each test, so even if tests are run in parallel, the sequence of random numbers within the test is deterministic.

This is to make sure tests are 100% reproducible and to guarantee that if the test stops passing, it's definitely down to code changes and not down to random noise.

File: test/test_braid.py [hunk 6]
```diff
@@ -521,3 +523,85 @@ def test_remove_sigma_inverse_pair_indices4(self):
...
+        for move in performable_moves:
+            states.append(move())
+
+        all_moves = self.get_all_moves(braid)
+
+        all_states = []       
+        for move in all_moves:
+            try:
+                state = move()
+            except IllegalTransformationException as e:
```
Comment (fhuszar) [inline test/test_braid.py:578]: Do you expect the error to be thrown here? generally try-except-pass seems suspicious.

File: test/test_braid.py [hunk 7]
```diff
@@ -521,3 +523,85 @@ def test_remove_sigma_inverse_pair_indices4(self):
...
+                pass
+            else:
+                all_states.append(state)
+        assert len(states) == len(all_states)
```
Comment (Error-42) [inline test/test_braid.py:582]: This shouldn't be here, because states may not be distinct.

**Review by Error-42 — CHANGES_REQUESTED** [review]

File: test/test_braid.py (hunk 5 above)
Comment (fhuszar) [reply test/test_braid.py:589]: Well done

**Pull request merged**